spec/system/frontend/solidus_stripe/checkout_spec.rb

```diff
@@ -2,218 +2,302 @@
 
 require 'solidus_stripe_spec_helper'
 
-RSpec.describe "Checkout with Stripe", :js do
+RSpec.describe 'Checkout', :js do
   include SolidusStripe::CheckoutTestHelper
 
-  it "completes as a registered user and reuses the payment using setup intents" do
-    create(:stripe_payment_method, preferred_stripe_intents_flow: 'setup')
-    visit_payment_step(user: create(:user))
-    choose_new_stripe_payment
-    fill_stripe_form
-    submit_payment
-    expect_payments_state(Spree::Order.last, ['checkout'])
-    confirm_order
-
-    order = Spree::Order.last
-    user = order.user
-    payment = order.payments.first
-    reusable_source = payment.source
-
-    expect(Spree::Order.count).to eq(1)
-    expect_checkout_completion(order)
-    expect_payments_state(order, ['pending'])
-    payment.capture!
-    expect_payments_state(order, ['completed'], outstanding: 0)
-    expect(SolidusStripe::PaymentSource.count).to eq(1)
-    expect(SolidusStripe::PaymentSource.last.stripe_payment_method_id).to be_present
-
-    # Pay with the newly created wallet source
-    visit_payment_step(user: user)
-    find_existing_payment_radio(user.wallet_payment_sources.first.id).choose
-    submit_payment
-    expect_payments_state(Spree::Order.last, ['checkout'])
-    confirm_order
-
-    order = Spree::Order.last
-    payment = order.payments.valid.first
-    expect(Spree::Order.count).to eq(2)
-    expect(order.user).to eq(user)
-    expect_checkout_completion(order)
-    expect_payments_state(order, ['pending'])
-    expect(order.payments.valid.count).to eq(1)
-    payment.capture!
-    expect_payments_state(order, ['completed'], outstanding: 0)
-    expect(SolidusStripe::PaymentSource.count).to eq(1)
-    expect(SolidusStripe::PaymentSource.last.stripe_payment_method_id).to be_present
-    expect(payment.source).to eq(reusable_source)
-  end
+  # To learn more about setup_future_usage in different contexts:
+  # - SetupIntents   : https://stripe.com/docs/payments/setup-intents#increasing-success-rate-by-specifying-usage
+  # - PaymentIntents : https://stripe.com/docs/payments/payment-intents#future-usage
+  ['', 'on_session', 'off_session'].each do |setup_future_usage|
```

Comment:
I guess it's there to be consumed when using the payment intent flow... could we use it only in that case? Otherwise, we have a redundant test for setup intents.